pii_recognition/evaluation/character_level_evaluation.py

```diff
@@ -97,3 +98,46 @@ def compute_entity_recalls_for_ground_truth(
         recalls.append(entity_dict)
 
     return recalls
+
+
+def compute_pii_detection_f1(
+    precisions: List[float],
+    recalls: List[float],
+    recall_threshold: Optional[float] = None,
+) -> float:
+    """Evaluate performance of PII detection with F1.
+
+    Rollup precisions and recalls to calculate F1 on boundary detection for PII
+    recognition. Recall thresholding is supported, if the system can recognise
+    a certain portion of an entity beyond the threshold, we will consider it a
+    success.
+
+    Args:
+        precisions: a list of entity precision values.
+        recalls: a list of entity recall values.
+        recall_threshold: a float between 0 and 1. Any value greater than or equals
+            to the threshold would be rounded up to 1.
+
+    Returns:
+        F1 score.
+    """
+    if recall_threshold:
+        if recall_threshold > 1.0 or recall_threshold < 0.0:
+            raise ValueError(
+                f"Invalid threshold! Recall threshold must between 0 and 1 "
+                f"but got {recall_threshold}"
+            )
+
+    if not precisions and not recalls:
+        raise ValueError("You are passing empty precisions and recalls lists!")
+    if not precisions:
+        raise ValueError("You are passing empty precisions list!")
+    elif not recalls:
+        raise ValueError("You are passing empty recalls list!")
+
+    if recall_threshold:
+        recalls = [1.0 if item >= recall_threshold else item for item in recalls]
+
+    ave_precision = sum(precisions) / len(precisions)
```

Comment:
nitpick. you could also use `np.mean(precisions)`

Reply:
That is intended because of mypy type check. `np.mean(precisions)` returns `numpy.float64` instead of `float`. I am not sure how mypy works on numpy, it may silent the type check for numpy related things may not. Regardless, to use numpy along with mypy, I feel some investigations would be needed.